backend/src/v2/driver/driver_test.go

```diff
@@ -1621,3 +1622,158 @@ func Test_extendPodSpecPatch_GenericEphemeralVolume(t *testing.T) {
+					ParameterValues: map[string]*structpb.Value{
+						"access_modes":       accessModes,
+						"pvc_name":           structpb.NewStringValue("my-pvc"),
+						"storage_class_name": structpb.NewStringValue("standard"),
+						"volume_name":        structpb.NewStringValue("volume-name"),
+					},
+				},
+			},
+			true,
+			"parameter size not provided",
+		},
+		{
+			"annotations provided",
+			args{
+				&pipelinespec.ExecutorInput_Inputs{
+					ParameterValues: map[string]*structpb.Value{
+						"access_modes": accessModes,
+						"pvc_name":     structpb.NewStringValue("my-pvc"),
+						"size":         structpb.NewStringValue("5Gi"),
+						"annotations": structpb.NewStructValue(
+							&structpb.Struct{
+								Fields: map[string]*structpb.Value{
+									"key1": structpb.NewStringValue("value1"),
+									"key2": structpb.NewStringValue("value2"),
+								},
+							},
+						),
+						"volume_name": structpb.NewStringValue("volume-name"),
+					},
+				},
+			},
+			false,
+			"",
+		},
+		{
+			"storage_class_name not provided",
+			args{
+				&pipelinespec.ExecutorInput_Inputs{
+					ParameterValues: map[string]*structpb.Value{
+						"access_modes": accessModes,
+						"pvc_name":     structpb.NewStringValue("my-pvc"),
+						"size":         structpb.NewStringValue("5Gi"),
+						"volume_name":  structpb.NewStringValue("volume-name"),
+					},
+				},
+			},
+			false,
+			"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pvc, err := buildPVC(tt.args.inputs)
+			if tt.wantErr {
+				assert.NotNil(t, err)
+				assert.Nil(t, pvc)
+				assert.Contains(t, err.Error(), tt.errMsg)
+			} else {
+				assert.Nil(t, err)
+				assert.NotNil(t, pvc)
+				// Test the PVC fields
+				if storageClassNameInput, ok := tt.args.inputs.ParameterValues["storage_class_name"]; ok {
+					assert.Equal(t, storageClassNameInput.GetStringValue(), *pvc.Spec.StorageClassName)
+				} else {
+					assert.Nil(t, pvc.Spec.StorageClassName)
+				}
+				if annotationsInput, ok := tt.args.inputs.ParameterValues["annotations"]; ok {
+					annotations := annotationsInput.GetStructValue().GetFields()
+					assert.Equal(t, len(annotations), len(pvc.Annotations))
+					for key, value := range annotations {
+						assert.Equal(t, value.GetStringValue(), pvc.Annotations[key])
+					}
+				}
+				if pvcNameInput, ok := tt.args.inputs.ParameterValues["pvc_name"]; ok {
+					assert.Equal(t, pvcNameInput.GetStringValue(), pvc.Name)
+				}
+				if pvcNameSuffixInput, ok := tt.args.inputs.ParameterValues["pvc_name_suffix"]; ok {
+					assert.Equal(t, pvc.Name, pvcNameSuffixInput.GetStringValue())
+				}
+				if sizeInput, ok := tt.args.inputs.ParameterValues["size"]; ok {
+					storage := pvc.Spec.Resources.Requests[k8score.ResourceStorage]
+					assert.Equal(t, sizeInput.GetStringValue(), storage.String())
+				}
+				assert.Equal(t, k8score.ReadWriteOnce, pvc.Spec.AccessModes[0])
+				assert.Equal(t, "volume-name", pvc.Spec.VolumeName)
```

Comment:
Can you please use `require` instead of `assert` so that the test fails fast and doesn't run the subsequent assertions?

Reply:
Do you want to use `require` for Nil and NonNil or everything?

Reply:
It looks like we can replace everything. Please correct me if I'm wrong.
I'm asking this because, in most cases, it doesn't make sense to continue the assertions when the first one fails. It only makes the logs harder to read and figure out what caused the failure due to too many failed assertions.

Reply:
@leseb that sounds good.
The repo doesn't have any policy regarding this.